src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodWithGCInfo.cs

```diff
@@ -34,6 +35,20 @@ public MethodWithGCInfo(MethodDesc methodDesc)
             _method = methodDesc;
         }
 
+        protected override void OnMarked(NodeFactory context)
+        {
+            // Once past phase 1, no new methods which are interesting for compilation may be marked except for methods
```

Comment:
Would there be a chance to validate some of these claims via assertions or some such?